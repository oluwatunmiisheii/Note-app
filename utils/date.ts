export function getFormattedDate(date: Date) {
  const month = date.getMonth() + 1;
  const day = date.getDate();
  const year = date.getFullYear();

  return `${year}-${month}-${day}`;
}