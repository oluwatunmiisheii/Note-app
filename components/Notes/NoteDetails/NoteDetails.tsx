import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { Note } from "../../../models/note.model";
import colors from "../../../utils/constants/colors";
import { getFormattedDate } from "../../../utils/date";

export const NoteDetails = ({ note }: { note: Note }) => {
  return (
    <View style={styles.container}>
      <View style={styles.item}>
        <Text style={styles.itemTitle}>Title</Text>
        <Text style={styles.itemBody}>{note.title}</Text>
      </View>
      <View style={styles.item}>
        <Text style={styles.itemTitle}>Description</Text>
        <Text style={styles.itemBody}>{note.description}</Text>
      </View>
      <View style={styles.item}>
        <Text style={styles.itemTitle}>Date</Text>
        <Text style={styles.itemBody}>{getFormattedDate(note.date)}</Text>
      </View>
      <View style={styles.item}>
        <Text style={styles.itemTitle}>Note</Text>
        <Text style={styles.itemBody}>{note.note}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    padding: 20,
    height: "100%",
    flexGrow: 1,
    flex: 1,
  },
  item: {
    marginBottom: 20,
  },
  itemTitle: {
    fontSize: 18,
    color: colors.gray,
    paddingBottom: 5,
  },
  itemBody: {
    fontSize: 20,
    color: "#000000",
  },
});
