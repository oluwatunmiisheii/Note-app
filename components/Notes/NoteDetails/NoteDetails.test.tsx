import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react-native";
import { getNote } from "../../../services/note/__mock_data__/notes.mock";
import { NoteDetails } from "./NoteDetails";
import { getFormattedDate } from "../../../utils/date";

const note = getNote();

const defaultProps: React.ComponentProps<typeof NoteDetails> = {
  note,
};

const create = (props = defaultProps) => {
  const tree = renderer.create(<NoteDetails {...props} />).toJSON();
  return { props, tree };
};

const setup = (props = defaultProps) => {
  const utils = render(<NoteDetails {...props} />);
  return { props, ...utils };
};

describe(`<${NoteDetails.name}>`, () => {
  test("renders correctly", () => {
    const { tree } = create();
    expect(tree).toMatchSnapshot();
  });

  test("renders note title", () => {
    setup();
    expect(screen.getByText(note.title)).toBeOnTheScreen();
  });

  test("renders note description", () => {
    setup();
    expect(screen.getByText(note.description)).toBeOnTheScreen();
  });

  test("renders note date", () => {
    setup();
    expect(screen.getByText(getFormattedDate(note.date))).toBeOnTheScreen();
  });

  test("renders note note", () => {
    setup();
    expect(screen.getByText(note.note)).toBeOnTheScreen();
  });
});
