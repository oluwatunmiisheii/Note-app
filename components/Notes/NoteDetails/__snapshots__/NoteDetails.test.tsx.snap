// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteDetails> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexGrow": 1,
      "height": "100%",
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#6c757d",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 20,
        }
      }
    >
      Note title
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#6c757d",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Description
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 20,
        }
      }
    >
      Note description
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#6c757d",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Date
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 20,
        }
      }
    >
      2023-2-15
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#6c757d",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Note
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 20,
        }
      }
    >
      Note note
    </Text>
  </View>
</View>
`;
